spec/Section 5 -- Validation.md

```diff
@@ -1855,6 +1855,26 @@ variable.
 
 ### All Variable Usages Are Allowed
 
+As GraphQL evolves and issues are spotted and fixed, the specification aims to
+maintain compatibility with existing GraphQL documents. In rare cases, this
+requires two versions of an algorithm, a Legacy Algorithm which supports clients
+with both old and new GraphQL documents, and a Strict Algorithm which only
+supports new GraphQL documents. It is recommended that you implement the Strict
+Algorithm unless you have existing GraphQL documents that you need to support
+which would become invalid under the Strict Algorithm, in which case you should
+implement the Legacy Algorithm.
+
+The All Variable Usages Are Allowed validation rule has become stricter since
+the release of the GraphQL Specification, specifically the Strict Algorithm no
+longer allows a nullable variable to be used in a non-nullable position even
+when either or both of the variable definition and the input position have a
+default value.
+
+#### Strict All Variable Usages Are Allowed
+
+Implement this only if you are not implementing
+[Legacy All Variable Usages Are Allowed](#sec-Legacy-All-Variable-Usages-Are-Allowed).
```

Comment:
I'm not sure "Implement this only if" is what we want. If you implement legacy then we *also* want you implementing strict.

Maybe we should have validation warnings? Where if you implement both:
- if strict validation succeeds, no error
- if legacy validation succeeds, warn
- if legacy validation fails, fail.

Reply:
Implementing both would be redundant, wouldn’t it? But yeah, this shouldn’t be a strict rule, probably a “we recommend you implement this only if” instead, from an efficiency POV.

Reply:
(I like the warn idea.)